driver-core/src/main/java/com/datastax/driver/core/Metadata.java

```diff
@@ -131,22 +127,56 @@ static String handleId(String id) {
         if (id == null)
             return null;
 
-        if (alphanumeric.matcher(id).matches())
+        if (isAlphanumeric(id))
             return id.toLowerCase();
 
         // Check if it's enclosed in quotes. If it is, remove them and unescape internal double quotes
         return ParseUtils.unDoubleQuote(id);
     }
 
+    private static boolean isAlphanumeric(String s) {
+        for (int i = 0; i < s.length(); i++) {
+            char c = s.charAt(i);
+            if (!(
+                    (c >= 48 && c <= 57) // 0-9
+                            || (c >= 65 && c <= 90) // A-Z
+                            || (c == 95) // _ (underscore)
+                            || (c >= 97 && c <= 122) // a-z
+            ))
+                return false;
+        }
+        return true;
+    }
+
     // Escape a CQL3 identifier based on its value as read from the schema
     // tables. Because it comes from Cassandra, we could just always quote it,
     // but to get a nicer output we don't do it if it's not necessary.
     static String escapeId(String ident) {
-        // we don't need to escape if it's lowercase and match non-quoted CQL3 ids,
-        // and if it's not a CQL reserved keyword
-        return lowercaseAlphanumeric.matcher(ident).matches()
-                && !isReservedCqlKeyword(ident) ?
-                ident : quote(ident);
+        return needsQuote(ident)
+                ? quote(ident)
+                : ident;
+    }
+
+    /**
+     * We don't need to escape an identifier if it
+     * matches non-quoted CQL3 ids ([a-z][a-z0-9_]*),
+     * and if it's not a CQL reserved keyword.
+     */
+    private static boolean needsQuote(String s) {
+        char c = s.charAt(0);
```

Comment:
It seems unlikely, but since `escapeId` is used in a ton of places, wouldn't this throw a `StringIndexOutOfBoundsException` if the input `s` is empty?